trieste/space.py

```diff
@@ -155,42 +155,58 @@ def __deepcopy__(self, memo: Dict[int, object]) -> DiscreteSearchSpace:
         return self
 
 
+Vector = TypeVar("Vector", tf.Tensor, List[float])
+""" A type variable representing either a :class:`tf.Tensor` or `list`. """
+
+
 class Box(SearchSpace):
     r"""
     Continuous :class:`SearchSpace` representing a :math:`D`-dimensional box in
     :math:`\mathbb{R}^D`. Mathematically it is equivalent to the Cartesian product of :math:`D`
     closed bounded intervals in :math:`\mathbb{R}`.
     """
 
-    def __init__(self, lower: TensorType, upper: TensorType):
-        """
-        :param lower: The lower (inclusive) bounds of the box.
-        :param upper: The upper (inclusive) bounds of the box.
-        :raise ValueError: If ``lower`` and ``upper`` have different shapes. Or if ``upper`` is not
+    def __init__(self, lower: Vector, upper: Vector):
```

Comment:
overloads would be preferable from a code point of view, but the docs don't render overloads for `__init__` so I've used a `TypeVar` instead